### DISCUSSION TITLE
Localized noise?

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#20135] [Q&A Modules: Phase field] [answered]
Hi,

I want to use something like `ConservedNormalNoise` but for only a specific part of my mesh (say between 0 < x < 10). Is there an easy way to do this?

I suppose I could modify the `ConservedLangevinNoise::computeQpResidual()` to be multiplied by a static variable that is only nonzero in this region but that seems like a bit of a hack.

Are there any easy examples to do this without the hack?

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use the `block` parameter for that. 
Make sure part of your mesh has a defined subdomain ID, using the moose mesh generators (https://mooseframework.inl.gov/syntax/Mesh/index.html) if necessary to assign it, then set block = 'that subdomain ID' in the ConservedNormalNoise UO parameters. 

Guillaume

**Reply by mangerij** [reply]
Hmm, I had thought about that. I guess I could try it. I was hoping this could be a general approach, I think this would only be restricted to hex elements. 

Thanks!

**Reply by GiudGiud** [reply]
I think blocks cannot be composed of mixed element types. Do you think that ll preclude this approach?

**Reply by mangerij** [reply]
I think the better way is to define a function such that the Kernel for noise overrides with

_function[_qp]*ConservedLangevinNoise::computeQpResidual()

where _function[_qp] = 1 by default but could in principle be a localized gaussian, cosine, sinc(), etc perturbation. This could also depend on time too which can be useful.

I went ahead with this approach and it works quite nicely :)
